src/jsonpath.ts

```diff
@@ -0,0 +1,197 @@
+      this.parse();
+    }
+
+    const list = {};
+
+    this.jsonpathMap.forEach((value, key) => {
+      const disclosure = this.digestMap.get(value);
+      list[key] = disclosure.value;
+    });
+
+    return list;
+  }
+
+  public getDisclosuresFromJsonpaths(disclosuresJsonpath: string[], opts?: getDisclosuresOptions) {
+    if (!this.parsed) {
+      this.parse();
+    }
+
+    const digests = disclosuresJsonpath.map((jsonpath) => {
+      const digest = this.jsonpathMap.get(jsonpath.startsWith('$.') ? jsonpath : '$.' + jsonpath);
+
+      if (!opts?.ignoreInvalid && !digest) {
+        throw new Error(`Cannot find disclosure for provided jsonpath ${jsonpath}`);
+      }
+
+      return digest;
+    });
+
+    const selectedDigests = new Set();
+
+    digests.forEach((digest) => {
+      selectedDigests.add(digest);
+      const parentDigests = this.getParentDigests(digest);
+      parentDigests.forEach((digest: string) => selectedDigests.add(digest));
+    });
+
+    const disclosures = [...selectedDigests].map((digest: string) => this.digestMap.get(digest)?.disclosure);
+
+    return disclosures;
+  }
+
+  private parseSdDigest(digest: string, parent = null, pathname = '$') {
+    const disclosure = this.digestMap.get(digest);
+    if (disclosure) {
+      if (parent) {
+        this.parentMap.set(digest, parent);
+      }
+
+      // if key is missing; disclosure is an array item;
+      const jsonpath = disclosure.key ? `${pathname}.${disclosure.key}` : pathname;
+      this.jsonpathMap.set(jsonpath, digest);
+
+      this.objectTraverse(disclosure.value, digest, jsonpath);
+    }
+  }
+
+  private objectTraverse(obj, parent = null, pathname = '$') {
+    if (obj instanceof Array) {
+      return this.arrayTraverse(obj, parent, pathname);
+    }
+
+    if (!isObject(obj)) {
+      return;
+    }
+
+    for (const key in obj) {
+      if (key !== SD_DIGEST && key !== SD_LIST_PREFIX && obj[key] instanceof Object) {
+        this.objectTraverse(obj[key], parent, `${pathname}.${key}`);
+      }
+    }
+
+    if (obj[SD_DIGEST]) {
+      obj[SD_DIGEST].forEach((digest) => {
+        this.parseSdDigest(digest, parent, pathname);
+      });
+    }
+  }
+
+  private arrayTraverse(arr, parent = null, pathname = '$') {
+    arr.forEach((item, index) => {
+      const jsonpath = `${pathname}[${index}]`;
+
+      if (item instanceof Array) {
+        return this.arrayTraverse(item, parent, jsonpath);
+      }
+
+      if (isObject(item) && !item[SD_LIST_PREFIX]) {
+        return this.objectTraverse(item, parent, jsonpath);
+      }
+
+      return this.parseSdDigest(item[SD_LIST_PREFIX], parent, jsonpath);
+    });
+  }
+
+  private getParentDigests(digest: string, collection: string[] = []) {
+    if (!digest) {
+      return collection;
+    }
+
+    const parent = this.parentMap.get(digest);
+
+    if (parent) {
+      collection.push(parent);
+    }
+
+    return this.getParentDigests(parent, collection);
+  }
+}
+
+export class SDJsonpath {
```

Comment:
Would be nice to have some consistency in naming of the classes/functions. Here `SD` is both capitals. In other cases it is written as `Sd`, `SdJwt`, `SDJWT`. Could you please pick one and update the codebase?